### PR TITLE
Remove DB Migration Runner

### PR DESCRIPTION
**Pull request by pauljamescleary**
DB Migrations (flyway) are run out-of-band.  They require you to
first `sbt assembly` and build a fat jar, and then subsequently
to run a `DbMigrationRunner`.

All of those steps are unnecessary as flyway supports concurrent
attempts to run migrations.  The specific item is in the FAQ
at https://flywaydb.org/documentation/faq#parallel

Instead, just run migrations when the application starts.

* Modified `VinylDNSJDBC` to _always_ run database migrations.  Removed
the embedded flag as that is no longer needed.
* Deleted the `migrations.conf` file as that will no longer be used.
* Cleaned up the config files.  The salient change was to remove the
`local-mode` flag, as it is no longer needed.  Found some left over
cruft along the way that is unused and cleared that up.

### CODE REVIEW

File: modules/api/src/main/scala/vinyldns/api/repository/mysql/VinylDNSJDBC.scala [hunk 1]
```diff
@@ -68,21 +61,19 @@ class VinylDNSJDBC(config: Config) {
   }
 
   // Only run migrations if we are using embedded db as embedded requires initialization
```
Comment (nimaeskandary) [inline modules/api/src/main/scala/vinyldns/api/repository/mysql/VinylDNSJDBC.scala]: Is this comment still relevant?

File: modules/api/src/it/resources/application.conf [hunk 2]
```diff
@@ -3,70 +3,9 @@ akka {
...
   log-dead-letters = 0
   logger-startup-timeout = 60s
-
-  actor {
-    provider = "akka.actor.LocalActorRefProvider"
-  }
 }
 
 vinyldns {
-  dns-backends = [
```
Comment (rebstar6) [inline modules/api/src/it/resources/application.conf]: @pauljamescleary where did we land on the multiple backends stream of work? we still are only at one, i assume all this stuff went in when we were working towards that.

do we have cards or something tracking that (even if low priority)?

File: modules/api/src/main/scala/vinyldns/api/repository/mysql/VinylDNSJDBC.scala (hunk 1 above)
Comment (rebstar6) [reply modules/api/src/main/scala/vinyldns/api/repository/mysql/VinylDNSJDBC.scala]: nope

File: modules/api/src/main/scala/vinyldns/api/repository/mysql/VinylDNSJDBC.scala [hunk 3]
```diff
@@ -68,21 +61,19 @@ class VinylDNSJDBC(config: Config) {
...
+  val dbName = config.getString("name")
+  val placeholders = Map("dbName" -> dbName)
+  migration.setPlaceholders(placeholders.asJava)
+  migration.setSchemas(dbName)
 
-    // Runs ALL flyway migrations including SQL and scala, assumes empty databases
-    migration.migrate()
-    logger.info("migrations complete")
-  }
+  // Runs ALL flyway migrations including SQL and scala, assumes empty databases
```
Comment (rebstar6) [inline modules/api/src/main/scala/vinyldns/api/repository/mysql/VinylDNSJDBC.scala]: nor is this comment. we need to make sure our migrations are fine to be run multiple times on non-empty databases if this is to work

File: docker/api/docker.conf [hunk 4]
```diff
@@ -50,7 +50,6 @@ vinyldns {
   # default settings point to the setup from docker compose
   db {
```
Comment (rebstar6) [inline docker/api/docker.conf]: docker/api/run.sh has
```
echo "Running migrations..."
java -Dconfig.resource=db-migrations.conf -cp /app/vinyldns-server.jar db.migration.MigrationRunner
```

need to remove

Comment (pauljamescleary) [reply docker/api/docker.conf]: good catch, surprised func tests pass

File: modules/api/src/it/resources/application.conf (hunk 2 above)
Comment (pauljamescleary) [reply modules/api/src/it/resources/application.conf]: we have cards in the old repo to track.  This needs some re-thinking.  I did not port over Deepak's code when we open sourced.

**Pull request merged**